pkg/spec/util.go

```diff
@@ -0,0 +1,215 @@
+		if val.Type().PkgPath() == kedgeSpecPackagePath {
+			for i := 0; i < val.NumField(); i++ {
+				field := val.Type().Field(i)
+
+				fieldJSONTags := strings.Split(field.Tag.Get("json"), ",")
+
+				// Adding fields that have "conflicting" as a JSON tag to the
+				// array. We need not check the first tag that is used for
+				// unmarshalling.
+				for _, tag := range fieldJSONTags[1:] {
+					if tag == "conflicting" {
+						blacklistedTags = append(blacklistedTags, fieldJSONTags[0])
+					}
+				}
+			}
+		}
+	}
+	return blacklistedTags, nil
+}
+
+// This function checks if the input parameter has the pointer to a struct as
+// its underlying type.
+// Returns "nil" if true, or an error if false
+func checkTypePointerToStruct(input interface{}) error {
+	if reflect.Indirect(reflect.ValueOf(input)).Kind().String() != "struct" {
+		return errors.New(fmt.Sprintf("Expected pointer to struct, got %v", reflect.ValueOf(input).Kind().String()))
+	}
+	return nil
+}
+
+// This function takes in pointer to a struct as input, and returns a map of
+// conflicting JSON tags used for unmarshaling.
+// All of the embedded structs are taken into account, and the fields with a
+// JSON tag "conflicting" are assumed to be handled.
+// The returned map contains the JSON tags as the keys, and the values are an
+// array of struct names which contain those fields, without being handled.
+func findConflictingJSONTags(inputStruct interface{}) (map[string][]string, error) {
+
+	// We need to check that no JSON tag is specified more than once at the
+	// top level of the struct.
+	// This means that the struct fields, and the JSON tags of all the
+	// fields of all the embedded structs need to be unique.
+	// We accomplish this over a 4 step process.
+
+	// Checking if input is pointer to struct
+	if err := checkTypePointerToStruct(inputStruct); err != nil {
+		return nil, errors.Wrap(err, "Input parameter type mismatch")
+	}
+
+	// Step 1: Get all the target structs.
+	// Look at tags in the fields of the input struct as well as the embedded
+	// structs, so both of them become our target structs.
+
+	var targetStructs []interface{}
+	// The first target struct is the input struct to the function
+	targetStructs = append(targetStructs, inputStruct)
+
+	// Getting all embedded structs for the input struct and appending to the
+	// list of target structs.
+	embeddedStructs, err := getEmbeddedStructs(inputStruct)
+	if err != nil {
+		return nil, errors.Wrapf(err, "Error getting embedded structs for struct: %v", inputStruct)
+	}
+	targetStructs = append(targetStructs, embeddedStructs...)
+
+	// Step 2: Get blacklisted tags.
+	// Get the fields which have already been handled and were previously
+	// conflicting. This needs to be done only for the structs in spec.go,
+	// and with fields marked as "conflicting" using a JSON tag
+
+	blacklistedTags, err := getMarkedAsConflictingJSONUnmarshalTags(targetStructs)
```

Comment:
if `blacklistedTags` could be map(of type `map[string]interface{}` value does not matter) it will be helpful in searching, and the usefulness of `blacklistedTags` is only to search.